pkg/products/reconciler.go

```diff
@@ -99,6 +109,9 @@ func NewReconciler(product integreatlyv1alpha1.ProductName, rc *rest.Config, con
 		tsClient := threescale.NewThreeScaleClient(httpc, installation.Spec.RoutingSubdomain)
 
 		reconciler, err = threescale.NewReconciler(configManager, installation, appsv1, oauthv1Client, tsClient, mpm, recorder)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
The linter complains with below if I remove. Do we care about lint errors?

pkg/products/reconciler.go:111:15: ineffectual assignment to `err` (ineffassign)
		reconciler, err = threescale.NewReconciler(configManager, installation, appsv1, oauthv1Client, tsClient, mpm, recorder)

Reply:
oh, maybe `err` needs to be declared before `switch`?

Reply:
The reason this is needed is because err is redeclared in the case statement. If you look at the whole function err is redeclared several times (in several case statements).
It seems err does not really need to be redeclared but the associated return types do. 
for example: oauthv1Client, err := (L98)
It's a bit strange. 

I think checking the err is required otherwise the final return in the function i.e. L123 could return nil for err when there was actually an err returned from threescale.NewReconciler(

Reply:
I would think that if we declare err on higher scope (before switch), this call `oauthv1Client, err := (L98)` would use that variable instead of declaring a local one. But I am not claiming that you are wrong, it's just unexpected for me, and I am no Go expert...
Anyway, I have no problem with leaving these the error checks in. :)

Reply:
I experimented a bit with this. Declaring err before the switch didn't work. 
In the below code oauthv1Client needs to be declared which forces the redeclaration of err.
At which point the original declaration is now out of scope and the new one is in scope for the given case statement. 

`oauthv1Client, err := oauthClient.NewForConfig(rc)`

I'm new to go so happy to be corrected here....